src/rdkafka_conf.h

```diff
@@ -423,6 +423,7 @@ struct rd_kafka_conf_s {
         int queue_backpressure_thres;
         int max_retries;
         int retry_backoff_ms;
+        int retry_backoff_max_ms;
```

Comment:
There need to be three additional things, 

First, check that retry.max.backoff.ms does not exceed retry.backoff.ms, inside rd_kafka_conf_warn. If that is the case, log a warning that we will always be doing constant backoff with interval retry.max.backoff.ms. See this Java comment for reference: https://github.com/apache/kafka/pull/14111/files#diff-252aed396717ac3d05ce228cb680e857f87c7456624b792c32d2db7fed461cbdR244

Second, inside rd_kafka_conf_finalize, add something like this:

```
if (!rd_kafka_conf_is_modified(conf, "topic.metadata.refresh.fast.interval.ms"))
                        conf->metadata_refresh_fast_interval_ms = conf->retry_backoff_ms; 
```

Update the documentation of topic.metadata.refresh.fast.interval.ms accordingly, to reflect that if it's not set, the default value will be the same as the value of retry.backoff.ms. 

Finally, back in rd_kafka_conf_warn, add a warning if `rd_kafka_conf_is_modified(conf, "topic.metadata.refresh.fast.interval.ms") && conf->metadata_refresh_fast_interval_ms > conf->retry_backoff_max_ms`. The warning text will be similar to the first point of this comment.

Reply:
Please test these things at least manually.